### PR TITLE
Fix default behavior for 401/403 http response codes

### PR DESCRIPTION
**Pull request by sungam3r**
This PR aligns behavior with server project that returns 401/403.

### CODE REVIEW

File: src/GraphQL.Client/GraphQLHttpClientOptions.cs [hunk 1]
```diff
@@ -52,8 +52,10 @@ public class GraphQLHttpClientOptions
 
     /// <summary>
     /// Delegate to determine if GraphQL response may be properly deserialized into <see cref="GraphQLResponse{T}"/>.
+    /// Note that compatible to the draft graphql-over-http spec GraphQL Server MAY return 4xx status codes (401/403, etc.)
+    /// with well-formed GraphQL response containing errors collection.
     /// </summary>
-    public Func<HttpResponseMessage, bool> IsValidResponseToDeserialize { get; set; } = r => r.IsSuccessStatusCode || r.StatusCode == HttpStatusCode.BadRequest;
+    public Func<HttpResponseMessage, bool> IsValidResponseToDeserialize { get; set; } = r => r.IsSuccessStatusCode || r.StatusCode == HttpStatusCode.BadRequest || r.Content.Headers.ContentType.MediaType == "application/graphql+json";
```
Comment (rose-a) [inline src/GraphQL.Client/GraphQLHttpClientOptions.cs:58]: This tries to parse all responses in which the server manages to set the content type header to `application/graphql+json` (including 5xx responses). Are u sure this is a "bulletproof" assumption?

I think we should at least catch parsing exceptions in [GraphQLHttpClient](https://github.com/graphql-dotnet/graphql-client/blob/758ede0e1773dafc9f9c662e495e89fe852a9786/src/GraphQL.Client/GraphQLHttpClient.cs#L128) and wrap those up in a [GraphQLHttpRequestException](https://github.com/graphql-dotnet/graphql-client/blob/89ba50d68e14c31910d4cf4e384dbe84135b0de8/src/GraphQL.Client/GraphQLHttpRequestException.cs) to pass the raw response content to the user (this probably won't hurt, anyway).

**Comment by rose-a** [comment]
What's missing, too, is that this client doesn't set the *Accept* request header as defined in the [spec draft](https://github.com/graphql/graphql-over-http/blob/main/spec/GraphQLOverHTTP.md#accept), so the server will only respond with `application/graphql+json` if you specify the accept header in your application code.

We should set it to `application/graphql+json; charset=utf-8, application/json; charset=utf-8` as recommended. I propose to do that in [GraphQLHttpRequest.ToHttpRequestMessage](https://github.com/graphql-dotnet/graphql-client/blob/758ede0e1773dafc9f9c662e495e89fe852a9786/src/GraphQL.Client/GraphQLHttpRequest.cs#L35).

File: src/GraphQL.Client/GraphQLHttpClientOptions.cs (hunk 1 above)
Comment (sungam3r) [reply src/GraphQL.Client/GraphQLHttpClientOptions.cs:58]: If content type is `application/graphql+json` then the response **can** be deserialized as a valid graphql response. 

Regarding 5xx error codes from server - see https://github.com/graphql/graphql-over-http/blob/main/spec/GraphQLOverHTTP.md#applicationgraphqljson :
изображение

**Comment by sungam3r** [comment]
OK, will do tomorrow.

File: src/GraphQL.Client/GraphQLHttpRequest.cs [hunk 2]
```diff
@@ -38,6 +39,9 @@ public virtual HttpRequestMessage ToHttpRequestMessage(GraphQLHttpClientOptions
         {
             Content = new StringContent(serializer.SerializeToString(this), Encoding.UTF8, options.MediaType)
         };
+        message.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/graphql+json"));
```
Comment (sungam3r) [inline src/GraphQL.Client/GraphQLHttpRequest.cs:42]: `MediaTypeWithQualityHeaderValue` does not accept `application/graphql+json; charset=utf-8`. It throws `System.FormatException : The format of value 'application/graphql+json; charset=utf-8' is invalid`. I added Accept-Charset header that should work the same.

**Review by rose-a — APPROVED** [review]
looks good to me..

**Pull request merged**